### PR TITLE
Add documentation for list converters, splitters #253 

### PR DESCRIPTION
**Pull request by jeremysolarz**
As mentioned in a comment in #253 there is no documentation for `listConverter` / `splitter`.

This PR fixes this.

### CODE REVIEW

File: doc/index.adoc [hunk 1]
```diff
@@ -228,9 +236,32 @@ Then, all you need to do is declare your field with the correct type and specify
...
+
+[source,java]
+----
+@Parameter(names = "-files", converter = FileConverter.class)
+List<File> files;
+----
+
+JCommander will split the string `file1,file2,file3` into `file1`, `file2`, `file3` and feed it one by one to the converter.
+
+For an alternative solution to parse a list of values see <<list-value>>.
```
Comment (cbeust) [inline doc/index.adoc]: values, see

File: doc/index.adoc [hunk 2]
```diff
@@ -228,9 +236,32 @@ Then, all you need to do is declare your field with the correct type and specify
...
 ----
 
-JCommander ships with a few common converters (e.g. one that turns a comma separated list into a List<String>).
+JCommander ships with a few common converters (for more info please see the implementations of `IStringConverter`).
+
+===== Note
+
+If a converter is used for a List field, JCommander tries to use this converter on the value provided.
+
+For example, suppose you provide a string representing a list of files:
```
Comment (cbeust) [inline doc/index.adoc]: This sentence can be joined with the one above.

File: doc/index.adoc [hunk 3]
```diff
@@ -228,9 +236,32 @@ Then, all you need to do is declare your field with the correct type and specify
...
+
+[source,bash]
+----
+$ java App -files file1,file2,file3
+----
 
-=== By factory
+And you declared your field as List and specified the converter as an attribute:
```
Comment (cbeust) [inline doc/index.adoc]: Backquotes around List.

File: doc/index.adoc [hunk 4]
```diff
@@ -312,6 +343,100 @@ Assert.assertEquals(a.hostPort.port.intValue(), 8080);
...
 Another advantage of using string converter factories is that your factories can come from a dependency injection framework.
 
+[[list-value]]
+=== Custom types - List value
+
+Use the `listConverter=` attribute of the `@Parameter` annotation and assign a custom `IStringConverter` implementation to convert a `String` into a `List` of values.
+
+==== By annotation
+
+If your application needs a list of complex types write a list type converter by implementing the same interface as before:
```
Comment (cbeust) [inline doc/index.adoc]: types, write.

File: doc/index.adoc [hunk 5]
```diff
@@ -312,6 +343,100 @@ Assert.assertEquals(a.hostPort.port.intValue(), 8080);
...
+----
+
+The parameter `file1,file2,file3` is given to the `listConverter` and will the properly processed.
+
+JCommander ships with a default converter for `String` values.
+
+
+=== Splitting
+
+Use the `splitter=` attribute of the `@Parameter` annotation and assign a custom `IParameterSplitter` implementation to handle how parameters are split in sub-parts
```
Comment (cbeust) [inline doc/index.adoc]: Missing period.

File: doc/index.adoc [hunk 6]
```diff
@@ -312,6 +343,100 @@ Assert.assertEquals(a.hostPort.port.intValue(), 8080);
...
+
+==== By annotation
+
+As already mentioned in <<single-value>>, by default, JCommander tries to split parameters for List field types on comma.
```
Comment (cbeust) [inline doc/index.adoc]: No need for "As already mentioned...".

Comment (cbeust) [inline doc/index.adoc]: Also, "commas" (plural) or "on the comma character".

**Comment by jeremysolarz** [comment]
@cbeust I understand your concern. I tried to make one part of the PR less verbose. If it's still too verbose for you please tell me what you don't like & I will fix that 😄

**Comment by jeremysolarz** [comment]
Hi @cbeust,

What do you say, how can I improve the PR to include the part about `listConverter`s / `splitter`s in the documentation.

Best,

JS

**Comment by cbeust** [comment]
Your changes are now live at http://jcommander.org

Thanks!

**Comment by jeremysolarz** [comment]
Great news!

Thanks @cbeust

**Pull request merged**